pydantic/schema.py

```diff
@@ -805,8 +806,8 @@ def field_singleton_schema(  # noqa: C901 (ignore complexity)
         # All values have the same type
         field_type = values[0].__class__
         f_schema['enum'] = list(values)
-
-    if lenient_issubclass(field_type, Enum):
+        add_field_type_to_schema(field_type, f_schema)
```

Comment:
```suggestion
        add_field_type_to_schema(field_type, f_schema)
        
```
Maybe preserve the pre-existing whitespace separating the `if` & the `elif` logic

Reply:
I don't agree since the whole block with all the elif is consistent. But it's subjective :)